lib/dradis/plugins/mapping_service.rb

```diff
@@ -0,0 +1,99 @@
+module Dradis
+  module Plugins
+    class MappingService
+      attr_accessor :integration, :rtp_id, :source
+
+      def initialize(args = {})
+        @integration   = args.fetch(:integration)
+        @rtp_id        = args.fetch(:rtp_id, nil)
+        @templates_dir = args.fetch(:templates_dir, default_templates_dir)
```

Comment:
"templates_dir" seems to belong to the old way of doing things, not the mappings / db way, no?

I see this is for dealing with the source fields and sample. But the source fields are a concern of the source, not the mapping service. And the sample, is also an integration concern, nothing we should be handling at this layer.

Reply:
This is still being used to pull the `.sample`, but I renamed it to `sample_dir` since `templates_dir` doesn't make sense anymore